exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/util/DsosMapperUtil.java

```diff
@@ -40,27 +45,39 @@
         public static Optional<Boolean> symptomsExist(int dsos) {
             DsosInterpretationMapper mapper = mapperInRange(dsos);
 
-            if (SYMPTOMS_UNKNOWN.equals(mapper)) {
+            if (SYMPTOMS_UNKNOWN.equals(mapper) || UNKNOWN.equals(mapper)) {
                 return Optional.empty();
             } else {
                 return Optional.of(!NO_SYMPTOMS.equals(mapper));
             }
         }
 
-        public static int mapToEfgs(Optional<Integer> localDsos, Optional<Boolean> symptomsExist) {
-            final int dsos = localDsos.orElse(DEFAULT_LOCAL_DAYS_SINCE_SYMPTOMS);
+        public static int mapToEfgs(TemporaryExposureKey localKey) {
+            final int dsos = localKey.daysSinceOnsetOfSymptoms.orElse(DEFAULT_LOCAL_DAYS_SINCE_SYMPTOMS);
+            final int submissionDsos = calculateDsosFromSubmission(localKey);
             AtomicInteger efgsDsos = new AtomicInteger(DEFAULT_LOCAL_DAYS_SINCE_SYMPTOMS);
```

Comment:
I think this could be done with just optional.map, without the atomicinteger & mutability
```
return localKey.symptomsExist
    .map(e -> mapDsos(dsos, submissionDsos, e))
    .orElseGet(() -> SYMPTOMS_UNKNOWN.zero + submissionDsos)
```

Reply:
Good point. I'll check that on net pr, where dsos nullability goes away.